xdb:restore doesn't support the "dir" attribute

### What is the problem

When using the ant task to restore a directory I get the following error:

```
BUILD FAILED
/home/mike/exist_ant_tasks/build.xml:28: xdb:restore doesn't support the "dir" attribute
```

### What did you expect
I'd expect the task to work and the data to be restored.

### Describe how to reproduce or add a test

According to the documentation "dir" is an acceptable attribute for the restore task:
http://www.exist-db.org/exist/apps/doc/ant-tasks.xml#D2.2.14

I'm using the following build file:

```
<?xml version="1.0" encoding="UTF-8"?>
<project default="Backup" xmlns:xdb="http://exist-db.org/ant">

    <!-- the properties file -->
    <property file="build.properties"/>

    <!-- classpath with eXist jars -->
    <path id="classpath.core">
        <fileset dir="${server.dir}/lib/core">
            <include name="*.jar"/>
        </fileset>
        <pathelement path="${server.dir}/exist.jar"/>
        <pathelement path="${server.dir}/exist-optional.jar"/>
    </path>

    <!-- include exist ant tasks -->
    <typedef resource="org/exist/ant/antlib.xml" uri="http://exist-db.org/ant">
        <classpath refid="classpath.core"/>
    </typedef>

    <!-- backup (collection) -->
    <target name="Backup" description="Backup collection to a directory">
        <xdb:backup uri="${backup.uri}" dir="${backup.dir}" user="${backup.user}" password="${backup.pass}"/>
    </target>

    <!-- restore (collection) -->
    <target name="Restore" description="Restore collection to a directory">
        <xdb:restore uri="${backup.uri}" user="${backup.user}" password="${backup.pass}" dir="${backup.dir}"/>
    </target>
</project>
```

Can Ant correctly convert an attribute value to a java.nio.file.Path, as used in the public setter method for the custom task?

https://github.com/eXist-db/exist/blob/develop/src/org/exist/ant/RestoreTask.java

### Context information

- 3.1.1 / fe05cf7527db943ae6f936c263f744e24fdae2c7
- Apache Ant(TM) version 1.9.7 compiled on April 9 2016
- openjdk version "1.8.0_131"
- OpenJDK Runtime Environment (build 1.8.0_131-b12)
- OpenJDK 64-Bit Server VM (build 25.131-b12, mixed mode)
- centos-release-7-3.1611.el7.centos.x86_64

--- a/src/org/exist/ant/RestoreTask.java
+++ b/src/org/exist/ant/RestoreTask.java
@@ -30,6 +30,7 @@ import org.exist.backup.Restore;
 import org.exist.backup.restore.listener.ConsoleRestoreListener;
 import org.exist.backup.restore.listener.RestoreListener;
 
+import java.io.File;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
@@ -164,15 +165,15 @@ public class RestoreTask extends AbstractXMLDBTask
      *
      * @param  dir
      */
-    public void setDir( Path dir )
+    public void setDir( File dir )
     {
-        this.dir = dir;
+        this.dir = dir.toPath();
     }
 
 
-    public void setFile( Path file )
+    public void setFile( File file )
     {
-        this.zipFile = file;
+        this.zipFile = file.toPath();
     }
 
 

--- a/test/src/ant/exist.properties
+++ b/test/src/ant/exist.properties
@@ -1,7 +1,8 @@
 exist.user=admin
 exist.password=
 exist.dir=../../..
-exist.uri=xmldb:exist:///
+exist.uri=xmldb:exist://
 test.coll=${exist.uri}/db
 test.file=test.xml
 test.binary.file=binary.gif
+test.backup-dir=./test/temp/backup

--- a/test/src/ant/server.xml
+++ b/test/src/ant/server.xml
@@ -5,12 +5,26 @@
     
     <import file="common.xml"/>
 
-    <target name="test_Backup">
-        <echo>test1</echo>
+    <target name="test_BackupRestore" depends="backup,restore"/>
+
+    <target name="backup">
+        <delete dir="${test.backup-dir}" failonerror="false"/>
+        <mkdir dir="${test.backup-dir}"/>
+        <echo>Creating backup ...</echo>
+        <xdb:backup initdb="true" uri="${test.coll}/antunit" dir="${test.backup-dir}" user="${exist.user}"
+                    password="${exist.password}"/>
+        <au:assertFileExists message="Backup task failed"
+                             file="${exist.dir}/test/temp/backup/db/antunit/__contents__.xml"/>
+        <xdb:remove uri="${test.coll}" user="${exist.user}" password="${exist.password}" collection="antunit"/>
     </target>
 
-    <target name="test_Restore">
-        <echo>test3</echo>
+    <target name="restore">
+        <echo>Restoring backup ...</echo>
+        <xdb:restore uri="${test.coll}" dir="${exist.dir}/test/temp/backup/db/antunit" user="${exist.user}"
+                     password="${exist.password}"/>
+        <au:assertTrue message="Restore did not work">
+            <xdb:exist uri="${test.coll}/antunit" resource="test1.xml"/>
+        </au:assertTrue>
     </target>
 
     <target name="test_XMLDBShutdown">

